docs/extensiondev.rst

```diff
@@ -542,3 +542,245 @@ your HTTP requests::
+        registry.add.assert_has_calls(calls, any_order=True)
+
+
+Testing backend actors
+----------------------
+
+Backends can usually be constructed with a small mockup of the configuration
+file, and mocking the audio actor::
+
+    @pytest.fixture
+    def config():
+        return {
+            'http': {
+                'hostname': '127.0.0.1',
+                'port': '6680'
+            },
+            'proxy': {
+                'hostname': 'host_mock',
+                'port': 'port_mock'
+            },
+            'my_extension': {
+                'enabled': True,
+                'param_1': 'value_1',
+                'param_2': 'value_2',
+                'param_n': 'value_n',
+            }
+        }
+
+    def get_backend(config):
+        return backend.MyBackend(config=config, audio=mock.Mock())
+
+The following libraries might be useful for mocking any HTTP requests that
+your extension makes:
+
+- `responses <https://pypi.python.org/pypi/responses>`_ - A utility library for
+  mocking out the requests Python library.
+- `vcrpy <https://pypi.python.org/pypi/vcrpy>`_ - Automatically mock your HTTP
+  interactions to simplify and speed up testing.
+
+At the very least, you'll probably want to patch ``requests`` or any other web
+API's that you use to avoid any unintended HTTP requests from being made by
+your backend during testing::
+
+    from mock import patch
+    @mock.patch('requests.get',
+                mock.Mock(side_effect=Exception('Intercepted unintended HTTP call')))
+
+
+Backend tests should also ensure that:
+
+- the backend provides a unique URI scheme,
+- that it sets up the various providers (e.g. library, playback, etc.)
+
+::
+
+    def test_uri_schemes(config):
+        backend = get_backend(config)
+
+        assert 'my_scheme' in backend.uri_schemes
+
+
+    def test_init_sets_up_the_providers(config):
+        backend = get_backend(config)
+
+        assert isinstance(backend.library, library.MyLibraryProvider)
+        assert isinstance(backend.playback, playback.MyPlaybackProvider)
+
+
+Once you have a backend instance to work with, testing the various playback,
+library, and other providers is straight forward and should not require any
+special setup or processing.
+
+Testing libraries
+-----------------
+
+Library test cases should cover the implementations of the standard Mopidy
+API (e.g. ``browse``, ``lookup``, ``refresh``, ``get_images``, ``search``,
+etc.)
+
+Testing playback controllers
+----------------------------
+
+Testing ``change_track`` and ``translate_uri`` is probably the highest
+priority, since these methods are used to prepare the track and provide its
+audio URL to Mopidy's core for playback.
+
+Testing frontends
+-----------------
+
+Because most frontends will interact with the Mopidy core, it will most likely
+be necessary to have a full core running for testing purposes::
+
+    self.core = core.Core.start(
+                config, backends=[get_backend(config]).proxy()
```

Comment:
There's a mix-up here in the parenthesis and the square bracket.
It should be "[get_backend(config).proxy()]"